forgerock-auth/src/main/java/org/forgerock/android/auth/callback/DeviceSigningVerifierCallback.kt

```diff
@@ -0,0 +1,191 @@
+    }
+
+    /**
+     * Bind the device.
+     *
+     * @param context  The Application Context
+     * @param listener The Listener to listen for the result
+     */
+    @JvmOverloads
+    fun sign(context: Context,
+             listener: FRListener<Void>) {
+        execute(context, listener = listener)
+    }
+
+    /**
+     * Helper method to execute signing, show biometric prompt.
+     *
+     * @param userKey User Information
+     * @param listener The Listener to listen for the result
+     * @param authInterface Interface to find the Authentication Type
+     */
+    protected open fun authenticate(userKey: UserKey,
+                                    listener: FRListener<Void>,
+                                    authInterface: Authenticator = getAuthenticator(userKey)) {
+
+        if(authInterface.isSupported().not()) {
+            handleException(Unsupported(), listener)
+            return
+        }
+        try {
+            authInterface.authenticate(timeout ?: 60) { result ->
+                if (result is Success) {
+                    val jws = authInterface.sign(userKey, challenge)
+                    setJws(jws)
+                    Listener.onSuccess(listener, null)
+                } else {
+                    // All the biometric exception is handled here , it could be Abort or timeout
+                    handleException(result, listener = listener)
+                }
+            }
+        }
+        catch (e: Exception) {
+            // This Exception happens only when there is Signing or keypair failed.
+            handleException(Unsupported(errorMessage = e.message), listener = listener)
+        }
+    }
+
+    /**
+     * Create the interface for the Authentication type(Biometric, Biometric_Fallback, none)
+     * @param context application or activity context
+     * @param userKeyService service to sort and fetch the keys stored in the device
+     * @param listener The Listener to listen for the result
+     */
+    protected open fun execute(context: Context,
+                               userKeyService: UserKeyService = UserDeviceKeyService(context),
+                               listener: FRListener<Void>) {
+
+        when(val status = userKeyService.getKeyStatus(userId)) {
+            is NoKeysFound -> handleException(UnRegister(), listener)
+            is SingleKeyFound -> authenticate(status.key , listener = listener)
+            else -> { showKeysFragment(InitProvider.getCurrentActivityAsFragmentActivity(), userKeyService, listener =  listener) }
+        }
+    }
+
+    /**
+     * Display fragment to select a user key from the list
+     * @param activity activity to be used to display the Fragment
+     * @param userKeyService service to sort and fetch the keys stored in the device
+     * @param fragment to display the users list
+     * @param listener The Listener to listen for the result
+     */
+    protected open fun showKeysFragment(activity: FragmentActivity,
```

Comment:
As a developer, I would like to override this method to provide my UI, I am thinking of how we can achieve that:
```public class MyCallback extends DeviceSigningVerifierCallback {
    @Override
    protected void showKeysFragment(@NonNull FragmentActivity activity, @NonNull UserKeyService userKeyService, @NonNull DialogFragment fragment, @NonNull FRListener<Void> listener) {
    }
}
```

Reply:
Would be good if we can allow developer to override with the following interface:
```java
    @Override
    protected void showKeysFragment(@NonNull FragmentActivity activity, @NonNull UserKeyService userKeyService ) {
        //somehow show my UI for all the userkeys and let the user pick one.
        UserKey userKey = userKeyService.getUserKeys().get(0);
        authenticate(userKey);
    }
```

Reply:
this is taken care